### PR TITLE
dmp-438 : Add Token Filter in Security Config

### PR DESCRIPTION
**Pull request by hemantasharma1129**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-438

### Change description ###
Added Token Filter in Security Config to check if Authorization Headers exist

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/application-local.yaml [hunk 1]
```diff
@@ -20,11 +20,26 @@ spring:
             prompt: login
             issuer-uri: https://hmctsdartsb2csbox.b2clogin.com/${AAD_B2C_TENANT_ID_KEY:00000000-0000-0000-0000-000000000000}/v2.0/
             provider: external-azure-ad-provider
+          internal-azure-ad:
```
Comment (richardbull) [inline src/main/resources/application-local.yaml:23]: I don't think we should be adding the same properties to local, dev profiles if they are already on the default `application.yaml`. We should only be adding the properties here that we want to override

Comment (hemantasharma1129) [reply src/main/resources/application-local.yaml:23]: This is a placeholder for internal AD and we will have new config/properties once justice team provides it.

**Review by cmagowan-hmcts — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/darts/authentication/AuthenticationFunctionalTest.java [hunk 2]
```diff
@@ -34,7 +35,7 @@ void shouldReturnUnauthorizedWhenSecuredEndpointIsCalledWithoutAuth() {
             .then()
             .extract().response();
 
-        assertEquals(401, response.statusCode());
+        assertFalse(response.getHeader("Location").isEmpty());
```
Comment (jackmaloney) [inline src/functionalTest/java/uk/gov/hmcts/darts/authentication/AuthenticationFunctionalTest.java:38]: Double negative makes this line abit hard to read - is there a better way? Do we know what the location should be or does it change each time? 
If not, maybe `assertTrue(response.getHeader("Location").isNotEmpty());` might be clearer?

Comment (hemantasharma1129) [reply src/functionalTest/java/uk/gov/hmcts/darts/authentication/AuthenticationFunctionalTest.java]: As discussed changed to assertNotnull.

**Pull request merged**